### PR TITLE
feat: support additional OIDC parameters

### PR DESCRIPTION
**Pull request by horus**
This commit allows some optional parameters to be passed along with the authentication request, including:

- Client secret: currently the dashboard has no support for the "confidential" type
- Additional scopes: for fine-grained access control

...could be a potential solution for the issue reported [here](https://asktug.com/t/topic/1004828).

Thank you!

### CODE REVIEW

**Comment by baurine** [comment]
hi @horus , thanks for your contribution! It will take some time to review.

**Comment by baurine** [comment]
hi @horus , can you sign the CLA? else the CI always fail.

File: ui/packages/tidb-dashboard-lib/src/apps/UserProfile/components/Form.SSO.tsx [hunk 1]
```diff
@@ -257,6 +257,27 @@ export function SSOForm() {
...
                 </Form.Item>
+                <Form.Item
+                  name="client_secret"
+                  label={t('user_profile.sso.form.client_secret')}
+                  rules={[{ required: false }]}
+                >
+                  <Input
+                    disabled={!isWriteable}
+                    style={DEFAULT_FORM_ITEM_STYLE}
+                  />
```
Comment (baurine) [inline ui/packages/tidb-dashboard-lib/src/apps/UserProfile/components/Form.SSO.tsx:268]: This should be showed as password, not plain text. I will help refine it later.

Comment (horus) [reply ui/packages/tidb-dashboard-lib/src/apps/UserProfile/components/Form.SSO.tsx:268]: > This should be showed as password, not plain text. I will help refine it later.

Yes, you are correct. Thanks for pointing it out.

**Comment by horus** [comment]
@baurine,

Just signed.
Again, thank you very much for your review and help.

**Pull request merged**